lua/lspconfig/configs.lua

```diff
@@ -100,7 +100,7 @@ function configs.__newindex(t, config_name, config_def)
 
     if config.autostart == true then
       local event_conf = config.filetypes and { event = 'FileType', pattern = config.filetypes }
-        or { event = 'BufReadPost' }
+        or { event = { 'BufReadPost', 'BufNewFile' } }
```

Comment:
`BufNewFile` run before `FileType`

Reply:
```vim
au FileType * echo 'FileType ' . &ft
au BufNewFile * echo 'BufNewFile ' . &ft
au BufReadPost * echo 'BufReadPost ' . &ft
```
Using the above for testing, both `BufNewFile` and `BufReadPost` run before `FileType`. Weirdly, when I run these commands with my usual config loaded, `FileType` actually executes first. I have no idea what part of my config changes the order.

If there's any issue with this callback triggering before `FileType`, it affects `BufReadPost` as well. I'm not sure if it matters here, since `BufReadPost` and `BufNewFile` are only used if `config.filetypes` is unset. Does it make sense for a server to be configured without `config.filetypes` if it matters what `filetype` is when `try_add()` runs?

At line 139 it's definitely important that `try_add_wrapper()` runs after `filetype` is set, however.

`vim.schedule()` seems to be enough to defer a function until after `FileType`, but this seems like a hacky solution to me.

Reply:
With the following in `test.lua`:
```lua
vim.api.nvim_create_autocmd({ "BufReadPost", "FileType", "BufNewFile" }, {
    callback = function(e)
        print(vim.o.ft, vim.inspect(e))
    end,
})
```
If I run `XDG_DATA_HOME=xdg_data XDG_STATE_HOME=xdg_state nvim -u test.lua`, `BufReadPost`/`BufNewFile` triggers first.
If I run `XDG_DATA_HOME=xdg_data XDG_STATE_HOME=xdg_state nvim -u NORC +source\ test.lua`, `FileType` triggers first.
If I wrap the entire contents of `test.lua` in `vim.schedule()`, `FileType` always triggers first regardless of whether I `:source` `test.lua` or load it as the rc file with `-u`.

`BufReadPost` and `BufNewFile` autocommands only seem to trigger earlier than `FileType` if they were set up sufficiently early during startup. This seems weird to me, but I can reproduce it consistently. I don't know what the intended execution order of these events is, or if there is even meant to be a guaranteed order at all.